grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -1209,4 +1209,78 @@ class QueryService implements Status {
         documents
     }
 
+     void updateOrspUserRoles (User user, ArrayList<String> newRoles) {
+        final session = sessionFactory.currentSession
+        final String query = ' insert into supplemental_role (version, role, user, user_id) values '
+        String queryAndValues = new StringBuffer()
+                .append(query)
+                .append(newRoles.withIndex().collect{ it, index ->
+            return "(0, :role${index}, :userName, :userId)"
+        }.join(" , ")).toString()
+
+        final SQLQuery sqlQuery = session.createSQLQuery(queryAndValues)
+        newRoles.eachWithIndex { String role, int i ->
+            sqlQuery.setString("role${i}", role)
+            sqlQuery.setString("userName", user.userName)
+            sqlQuery.setLong("userId", user.id)
+        }
+        sqlQuery.executeUpdate()
+    }
```

Comment:
This implementation seems very complex to me. It looks like it should be drop dead simple ... 
```
final String query = ' insert into supplemental_role (version, role, user, user_id) values (:version, :role, :user, :userId)'
newRoles.each { it -> 
  sqlQuery.setInteger("version", 0)
  sqlQuery.setString("role", it)
  sqlQuery.setString("userName", user.userName)
  sqlQuery.setString("userId", user.id)
  sqlQuery.executeUpdate()
} 
```

Reply:
Cool, My idea was to not execute n times a database query. But now that i think of it, newRoles never is going to be larger than 2. This is now changed